tests/client/test_client.py

```diff
@@ -68,18 +69,71 @@ def test_can_search_distributor(client, requests_mocker):
                 "id": "yum_distributor",
                 "distributor_type_id": "yum_distributor",
                 "repo_id": "test_rpm",
+                "config": {"relative_url": "relative/path"},
+            },
+            {
+                "id": "cdn_distributor",
+                "distributor_type_id": "rpm_rsync_distributor",
+                "config": {"relative_url": "relative/path"},
             },
-            {"id": "cdn_distributor", "distributor_type_id": "rpm_rsync_distributor"},
         ],
     )
 
     distributors_f = client.search_distributor()
     distributors = [dist for dist in distributors_f.result().as_iter()]
     # distributor objects are returned
     assert sorted(distributors) == [
-        Distributor(id="cdn_distributor", type_id="rpm_rsync_distributor"),
         Distributor(
-            id="yum_distributor", type_id="yum_distributor", repo_id="test_rpm"
+            id="cdn_distributor",
+            type_id="rpm_rsync_distributor",
+            relative_url="relative/path",
+        ),
+        Distributor(
+            id="yum_distributor",
+            type_id="yum_distributor",
+            repo_id="test_rpm",
+            relative_url="relative/path",
+        ),
+    ]
+    # api is called once
+    assert requests_mocker.call_count == 1
+
+
+def test_can_search_distributors_with_relative_url(client, requests_mocker):
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/distributors/search/",
+        json=[
+            {
+                "id": "yum_distributor",
+                "distributor_type_id": "yum_distributor",
+                "repo_id": "test_rpm",
+                "config": {"relative_url": "relative/path"},
+            },
+            {
+                "id": "cdn_distributor",
+                "distributor_type_id": "rpm_rsync_distributor",
+                "config": {"relative_url": "relative/path"},
+            },
+        ],
+    )
+
+    crit = Criteria.with_field("relative_url", Matcher.regex("relative/path"))
```

Comment:
This test would have been a little stronger if the test data and matcher were set up so this actually had to be a regex comparison.  Right now the test would pass if the regex was not evaluated as a regex but just an exact string comparison, since exactly the same values were chosen for the pattern and for the data being tested.